src/seismometer/data/loader/prediction.py

```diff
@@ -71,6 +71,57 @@ def _rename_targets(config: ConfigProvider, dataframe: pd.DataFrame) -> pd.DataF
 
 
 # post loaders
+def dictionary_types(config: ConfigProvider, dataframe: pd.DataFrame) -> pd.DataFrame:
+    """
+    Convert the loaded predictions dataframe to the expected types.
+
+    Prioritizes the types defined in the configuration dictionary, then falls back to assumed_types.
+
+
+    Parameters
+    ----------
+    config : ConfigProvider
+        The loaded configuration object.
+    dataframe : pd.DataFrame
+        The loaded predictions dataframe.
+
+    Returns
+    -------
+    pd.DataFrame
+        The predictions dataframe with adjusted types.
+
+    Raises
+    ------
+    ConfigurationError
+        If any columns cannot be converted to the expected types.
+    """
+
+    defined_types = _gather_defined_types(config)
+    unspecified_columns = []
+    value_error_columns = []
+
+    for col in dataframe.columns:
+        if col in defined_types:
+            try:
+                if "int" in defined_types[col].lower():
+                    dataframe[col] = dataframe[col].astype(float)
+                dataframe[col] = dataframe[col].astype(defined_types[col])
+            except ValueError:
+                value_error_columns.append(col)
+        else:
+            unspecified_columns.append(col)
+
+    if len(value_error_columns) > 0:
+        raise ConfigurationError(
+            f"Could not convert columns to expected types: {', '.join(value_error_columns)}. "
+            + "Update dictionary config or contact the model owner columns."
+        )
```

Comment:
Do we still want this, it looks like its also in pandas_helpers.post_process_event